### ISSUE TITLE
ModuleNotFoundError: No module named 'jsonschema.protocols'

### ISSUE THREAD

**Issue by lucasrodes** [owid/etl#708]
I just fresh-installed all dependencies and I am getting a `ModuleNotFoundError: No module named 'jsonschema.protocols'` error when running `make lab`.

* python version: 3.11.1 (also happens in 3.10.9)
* Poetry version: 1.3.1

I checked and seems that the submodule `jsonschema.protocols` is not installed.

Complete error message:

```
Traceback (most recent call last):
  File "/home/lucas/repos/etl/.venv/bin/jupyter-lab", line 5, in <module>
    from jupyterlab.labapp import main
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyterlab/labapp.py", line 13, in <module>
    from jupyter_server.serverapp import flags
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyter_server/serverapp.py", line 33, in <module>
    from jupyter_events.logger import EventLogger
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyter_events/__init__.py", line 2, in <module>
    from .logger import EVENTS_METADATA_VERSION, EventLogger
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyter_events/logger.py", line 18, in <module>
    from .schema_registry import SchemaRegistry
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyter_events/schema_registry.py", line 3, in <module>
    from .schema import EventSchema
  File "/home/lucas/repos/etl/.venv/lib/python3.11/site-packages/jupyter_events/schema.py", line 6, in <module>
    from jsonschema.protocols import Validator
ModuleNotFoundError: No module named 'jsonschema.protocols'
```

**Comment by lucasrodes** [comment]
When I install all dependencies (`poetry install`), version 3.2.0 of `jsonschema` library gets installed on my end. Checking in the [source code](https://github.com/python-jsonschema/jsonschema/tree/v3.2.0/jsonschema), it seems that `jsonschema.protocols` did not exist back then.

Seems that some changes were done to the dependencies affecting `jsonschema` in https://github.com/owid/etl/commit/ec009f0a1a8205e025d288a3b37b2cd8d7968e92

**Comment by lucasrodes** [comment]
Seems that this is fixed by running `poetry update jsonschema`, which update poetry.lock file and installs a more up-to-date jsonschema version.

**Comment by lucasrodes** [comment]
related threads in slack: 
- https://owid.slack.com/archives/C0193RW5E2J/p1672146738692139
- https://owid.slack.com/archives/C0193RW5E2J/p1672160077322289

**Comment by lucasrodes** [comment]
I think this gets solved if I change: `jsonschema = "^3.2.0"` to `jsonschema = ">=3.2.0"` in both [walden](https://github.com/owid/walden/blob/3cf02bf99a1e7143b4e256a4b56361a0685ea919/pyproject.toml#L10) and [owid-catalog-py](https://github.com/owid/owid-catalog-py/blob/53d31a7dd62ce05747b065222fbb49521bb967bf/pyproject.toml#L15) dependencies. In both cases, `jsonschema = "^3.2.0"` has been there for a while, so unsure what broke the environment. A jupyter-related library update? I have checked, but the last dependency changes concern `pyarrow`, `owid-repack`, `hydra-core`, `dvc`, etc.

**Comment by larsyencken** [comment]
Confirmed it as well on Ubuntu 22.04 arm64.

As part of trying to resolve library issues, I ran `poetry update` on each of the libraries which tries to do a new set of dependency resolutions on them. That's probably the cause of the updates.
